frontend/src/Connection.ts

```diff
@@ -334,13 +334,13 @@ export class Connection {
         }
 
         case ACTIONS.AddedChain: {
-          const [label, nodeCount] = message.payload;
-          const chain = chains.get(label);
+          const [label, genesisHash, nodeCount] = message.payload;
+          const chain = chains.get(genesisHash);
 
           if (chain) {
             chain.nodeCount = nodeCount;
           } else {
-            chains.set(label, { label, nodeCount });
+            chains.set(genesisHash, { label, genesisHash, nodeCount });
```

Comment:
If I have this correct the `label` is still being used to `comparePinnedChains`?